File: backend/app/api/notifications.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.deps import get_current_user, get_db
from app.models.user import User
from app.schemas.notification import NotificationListResponse, NotificationResponse
from app.services.notification import NotificationService

router = APIRouter()


@router.get("", response_model=NotificationListResponse)
async def list_notifications(
    user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    service = NotificationService(db)
    return await service.list(user)


@router.patch("/{notification_id}/read")
async def mark_as_read(
    notification_id: int,
    user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    service = NotificationService(db)
    return await service.mark_read(user, notification_id)

File: frontend/src/api/notifications.ts
import { apiClient } from './client'
import type { Notification, NotificationListResponse } from '@/types/notification'

export const notificationsApi = {
  // Получить список уведомлений
  list: async (): Promise<NotificationListResponse> => {
    const response = await apiClient.get('/notifications')
    return response.data
  },

  // Отметить уведомление как прочитанное
  markAsRead: async (notificationId: number): Promise<Notification> => {
    const response = await apiClient.patch(`/notifications/${notificationId}/read`)
    return response.data
  },
}